### DISCUSSION TITLE
How do I trigger from outside the scope?

### DISCUSSION THREAD

**Discussion by bakbuz** [alpinejs/alpine#651] [4. General]
Hello guys! I have a question.

How do I trigger from outside the scope?

Example:
https://codepen.io/bakbuz/pen/abdjExQ?editors=1011

Best regards

**Comment by KevinBatdorf** [comment]
That's in the docs under `$dispatch`

https://github.com/alpinejs/alpine#dispatch

Essentially you fire a custom event on one component and listen for it on another
```html
<button x-data @click="$dispatch('custom-event', 'Hello World!')">
<div x-data @custom-event.window="console.log($event.detail)"></div>
```

**Reply by bakbuz** [reply]
That's worked. Thank you very much.
